### DISCUSSION TITLE
Control Interface

### DISCUSSION THREAD

**Discussion by MichaelM4** [keirf/flashfloppy#703] [Ideas]
Hello,

I downloaded the source code from git and am pouring through it.  I am interested in adding a bit bang SPI for communication between the host system and flash floppy.  For example using step as SPICLK, direction and MOSI, and Track 0 status as MISO.

Any suggestions on where I should focus my efforts?

Thanks,
Mike

**Comment by keirf** [comment]
So you want to do synchronous duplex serial comms over the floppy interface?

**Comment by MichaelM4** [comment]
It does not need to be synchronous but I figured that would work better than trying to get timing correct on various platforms.

The idea is the host (TRS-80 Model 4 for example) can request a list of drive images on the usb stick and allow the user to select one to mount in flash floppy.

**Reply by keirf** [reply]
Maybe extend the existing direct access protocol on tracks 254 and 255. See image/da.c

**Reply by MichaelM4** [reply]
Excellent, I will have a look at that.

**Comment by MichaelM4** [comment]
Can you point me to a DA protocol document?

**Reply by keirf** [reply]
Google hxc direct access mode

**Comment by MichaelM4** [comment]
Got it. Thanks.

**Reply by MichaelM4** [reply]
Just need to add FindFirst and FindNext commands.
